common/dynamicconfig/collection_test.go

```diff
@@ -319,11 +284,106 @@ func (s *collectionSuite) TestGetMapProperty() {
 	s.Equal("321", value()["testKey"])
 }
 
-func (s *collectionSuite) TestGetIntPropertyFilteredByDestination() {
-	setting := DestinationIntSetting{
-		key: testGetIntPropertyFilteredByDestinationKey,
-		def: 10,
+func (s *collectionSuite) TestGetTyped() {
+	type myFancyType struct {
+		Number int
+		Names  []string
 	}
+	setting := NewGlobalTypedSettingWithConverter(
+		testGetTypedPropertyKey,
+		ConvertStructure(myFancyType{-3, nil}), // used if convert is called
+		myFancyType{28, []string{"global", "typed", "setting"}},
+		"",
+	)
+	get := setting.Get(s.cln)
+
+	s.Run("Default", func() {
+		s.Equal(setting.def, get())
+	})
+
+	s.Run("Basic", func() {
+		// map[string]any is what the yaml library decodes arbitrary data into
+		s.client[testGetTypedPropertyKey] = map[string]any{
+			"Number": 39,
+			"Names":  []string{"new", "names"},
+		}
+		s.Equal(myFancyType{
+			Number: 39,
+			Names:  []string{"new", "names"},
+		}, get())
+	})
+
+	s.Run("CaseInsensitive", func() {
+		s.client[testGetTypedPropertyKey] = map[string]any{
+			"naMES": []string{"case", "insensitive"},
+		}
+		s.Equal(-3, get().Number) // note the convert default is used here
+		s.Equal([]string{"case", "insensitive"}, get().Names)
+	})
+
+	s.Run("WrongType", func() {
```

Comment:
When this happens, do we log the error?

Reply:
We do, at a throttled level (first 1000 and then 1/1000 after).

Actually re-reading that path made me realize that we should simplify some of that logic. I'll follow up in another PR since it has nothing to do with typed settings really